go.mod

```diff
@@ -31,10 +31,10 @@ require (
 	google.golang.org/api v0.138.0
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.26.4
-	k8s.io/apimachinery v0.26.4
-	k8s.io/client-go v0.26.4
-	sigs.k8s.io/controller-runtime v0.14.6
+	k8s.io/api v0.28.0
```

Comment:
We still cannot make this change:
https://jira.mongodb.org/browse/CLOUDP-185089
https://github.com/operator-framework/operator-sdk/pull/6514
https://github.com/operator-framework/operator-sdk/issues/6381

@dependabot close